### PR TITLE
Hca rjsf chapter 1 remaining pages

### PR DESCRIPTION
**Pull request by annekainicUSDS**
Remaining 3 pages of Chapter 1 (merging it into the original PR will, I think, make it easier to review this one while the other PR is still open. Connected to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2230.

#### Demographic Information
screen shot 2017-05-01 at 4 53 15 pm

#### Veteran Address
screen shot 2017-05-01 at 4 53 31 pm

#### Contact Information
screen shot 2017-05-01 at 4 53 45 pm

### CODE REVIEW

File: src/js/hca-rjsf/config/form.js [hunk 1]
```diff
@@ -147,22 +163,146 @@ const formConfig = {
...
+          },
+          schema: {
+            type: 'object',
+            properties: {
+              address: _.merge(address.schema(true), {
+                properties: {
+                  street: {
+                    type: 'string',
+                    minLength: 1,
+                    maxLength: 30
```
Comment (annekainicUSDS) [inline src/js/hca-rjsf/config/form.js]: Current HCA has these maxLenghths for certain address fields due to ES requirements.

File: test/hca-rjsf/config/veteranInformation.unit.spec.jsx [hunk 2]
```diff
@@ -34,4 +34,46 @@ describe('HCA veteranInformation', () => {
     expect(formDOM.querySelectorAll('input, select').length).to.equal(6);
     expect(formDOM.querySelector('#root_veteranSocialSecurityNumber')).not.to.be.null;
   });
+
+  it('should render demographicInformation page', () => {
```
Comment (jbalboni) [inline test/hca-rjsf/config/veteranInformation.unit.spec.jsx]: We should have some tests for making sure you get the right number of required field errors when submitting on these pages.

**Review by jbalboni — APPROVED** [review]
Looks good, just needs a couple more tests, I think.

File: test/hca-rjsf/config/veteranInformation.unit.spec.jsx (hunk 2 above)
Comment (annekainicUSDS) [inline test/hca-rjsf/config/veteranInformation.unit.spec.jsx]: Good point, will add that!

**Comment by annekainicUSDS** [comment]
Added tests for required fields for all pages in this chapter.

File: src/js/common/schemaform/Address.jsx [hunk 3]
```diff
@@ -171,6 +171,20 @@ class Address extends React.Component {
             formContext={formContext}
             onChange={(update) => {this.handleChange('street2', update);}}
             onBlur={onBlur}/>
+
+        {schema.properties.street3 ? <SchemaField
+            name="street3"
+            required={this.isRequired('street3')}
+            schema={schema.properties.street3}
+            uiSchema={uiSchema.street3}
```
Comment (cvalarida) [inline src/js/common/schemaform/Address.jsx]: I'm not sure that it'll _really_ matter, but in `schemaform/definitions/address.js`, could we add a default `uiSchema.street3` like the other fields have? If nothing else, when we look at the definition file it'll act as an indication that we can, in fact, have 3 street lines if necessary.

File: test/hca-rjsf/config/veteranInformation.unit.spec.jsx [hunk 4]
```diff
@@ -2,36 +2,107 @@ import React from 'react';
...
+      <DefinitionTester
+          schema={schema}
+          data={{}}
+          onSubmit={onSubmit}
+          uiSchema={uiSchema}/>
+    );
+    const formDOM = findDOMNode(form);
+
+    expect(formDOM.querySelectorAll('input, select').length).to.equal(8);
+    expect(formDOM.querySelector('#root_gender')).not.to.be.null;
```
Comment (cvalarida) [inline test/hca-rjsf/config/veteranInformation.unit.spec.jsx]: Why is this here? Same with `#root_veteranFullName_first` above and `#root_address_country` below.

**Review by cvalarida — APPROVED** [review]
Looks good. Just had a couple questions.

File: test/hca-rjsf/config/veteranInformation.unit.spec.jsx (hunk 4 above)
Comment (annekainicUSDS) [inline test/hca-rjsf/config/veteranInformation.unit.spec.jsx]: This is a pattern we were using in the 1990n tests, which is where I think I started from in creating the HCA tests. It just checks that the first field for that page is actually rendered. It's a bit more specific to this page than just the number of inputs and selects. Open to feedback if there is a better way to test this.

Comment (cvalarida) [inline test/hca-rjsf/config/veteranInformation.unit.spec.jsx]: Nah, I think it's fine. Just something I hadn't seen on any of the edu forms and was wondering if we should be doing that there.

File: src/js/common/schemaform/Address.jsx (hunk 3 above)
Comment (annekainicUSDS) [inline src/js/common/schemaform/Address.jsx]: Yep, I can add that

File: src/js/common/schemaform/definitions/address.js [hunk 5]
```diff
@@ -48,6 +48,9 @@ export function uiSchema(label = 'Address') {
     street2: {
       'ui:title': 'Line 2'
     },
+    street3: {
```
Comment (annekainicUSDS) [inline src/js/common/schemaform/definitions/address.js]: @cvalarida per your suggestion added this here :)
